src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -284,52 +300,88 @@ protected override void StopProcessing()
         #region ConnectionTest
 
         private void ProcessConnectionByTCPPort(string targetNameOrAddress)
-        {
+        {    
             if (!TryResolveNameOrAddress(targetNameOrAddress, out _, out IPAddress? targetAddress))
             {
                 return;
             }
 
-            TcpClient client = new TcpClient();
+            int successfulConnections = 0;
 
-            try
+            for (var i = 1; i <= Count; i++)
             {
-                Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
-                string targetString = targetAddress.ToString();
-
-                for (var i = 1; i <= TimeoutSeconds; i++)
+                TcpTestStatus testResult = new TcpTestStatus(
+                    i,
+                    Source,
+                    targetNameOrAddress,
+                    targetAddress.ToString(),
+                    TcpPort,
+                    0,
+                    TcpTestStatus.TcpConnectionTestResult.New
+                );
+                
+                try
                 {
-                    Task timeoutTask = Task.Delay(millisecondsDelay: 1000);
+                    TcpClient client = new TcpClient();
+                    
+                    Stopwatch stopwatch = new Stopwatch();
+                    stopwatch.Start();
+
+                    Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
+                    Task timeoutTask = Task.Delay(new TimeSpan(0, 0, TimeoutSeconds));
+
                     Task.WhenAny(connectionTask, timeoutTask).Result.Wait();
+                    stopwatch.Stop();
```

Comment:
It must be in finally block.

Reply:
I might be interpreting things a bit differently. My reasoning behind stopping the stopwatch right after the task completion is to stop timing the latency as soon as the connection is completed. Even though it's a marginal difference, moving this to the finally block may add a bit of overhead, and skew the results a bit. 

Do you reckon there may be a better way to get this measurement? Thanks!

Reply:
If an exception throws we should stop the timer.
I suggest to move out `new Stopwatch()` out of the cycle and use `timer.Reset()` in finally block.

Reply:
Ah great point! I'll make that change!